src/main/java/com/direwolf20/buildinggadgets/client/gui/materiallist/MaterialListGUI.java

```diff
@@ -0,0 +1,214 @@
+package com.direwolf20.buildinggadgets.client.gui.materiallist;
+
+import com.direwolf20.buildinggadgets.client.gui.DireButton;
+import com.direwolf20.buildinggadgets.client.util.AlignmentUtil;
+import com.direwolf20.buildinggadgets.client.util.RenderUtil;
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.items.Template;
+import com.direwolf20.buildinggadgets.common.tools.InventoryManipulation;
+import com.direwolf20.buildinggadgets.common.tools.UniqueItem;
+import com.google.common.collect.Multiset;
+import it.unimi.dsi.fastutil.ints.IntList;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import org.lwjgl.input.Mouse;
+
+import java.awt.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is adapted from Lunatrius's Schematica mod, 1.12.2 version.
+ * <a href="github.com/Lunatrius/Schematica/blob/master/src/main/java/com/github/lunatrius/schematica/client/gui/control/GuiSchematicMaterials.java">Github</a>
+ */
+
+public class MaterialListGUI extends GuiScreen {
+
+    public static final int BUTTON_HEIGHT = 20;
+    public static final int BUTTONS_PADDING = 4;
+
+    public static final ResourceLocation BACKGROUND_TEXTURE = new ResourceLocation(BuildingGadgets.MODID, "textures/gui/material_list.png");
+    public static final int BACKGROUND_WIDTH = 256;
+    public static final int BACKGROUND_HEIGHT = 200;
+    public static final int BORDER_SIZE = 4;
+
+    public static final int WINDOW_WIDTH = BACKGROUND_WIDTH - BORDER_SIZE * 2;
+    public static final int WINDOW_HEIGHT = BACKGROUND_HEIGHT - BORDER_SIZE * 2;
+
+    private static List<ItemStack> toInternalMaterialList(Multiset<UniqueItem> templateList) {
+        List<ItemStack> result = new ArrayList<>();
+        for (Multiset.Entry<UniqueItem> entry : templateList.entrySet()) {
```

Comment:
```suggestion
        return templateList.entrySet().stream().map(e -> new ItemStack(e.getElement().item, e.getCount(), e.getElement().meta)).collect(Collectors.toList());
```
In multiple lines of course... You might also consider adding a toItemstack Method to UniqueItem...